internal/bsr/bsr.go

```diff
@@ -237,6 +238,173 @@ func OpenSession(ctx context.Context, sessionRecordingId string, f storage.FS, k
 	return session, nil
 }
 
+type Validation struct {
+	Name               string
+	Type               ContainerType
+	Error              error
+	ChecksumValidation ChecksumValidation
+	SubContainer       []Validation `json:"sub_container,omitempty"`
+}
+
+type ValidationSummary struct {
+	SessionRecordingId string
+	Valid              bool
+	FailedChecksums    map[ContainerType]map[string]ChecksumValidation
+}
+
+// Validate retrieves a BSR from storage using the sessionRecordingId and validates the BSR.
+// All files and sub container files will be verified by comparing it against checksums for
+// each file in SHA256SUM file.
+//
+// Validation will continue even if there's an error encountered during validation.
+// The validation error will be added to the Validation struct Error field.
+func Validate(ctx context.Context, sessionRecordingId string, f storage.FS, keyUnwrapFn kms.KeyUnwrapCallbackFunc) (Validation, ValidationSummary, error) {
```

Comment:
I don't think we need a separate struct for the ValidationSummary, at least not one that is completely separate from the Validation. Essentially I think one should contain the other, and also some details currently in the summary are not needed, at least as part of the bsr package.

I think instead we want something like:

```go
// Validation provides the results from validating a bsr.
type Validation struct {
  SessionRecordingId         string
  Valid                      bool
  SessionRecordingValidation *ContainerValidation
}

// ContainerValidation contains the results from validating a container in a bsr.
type ContainerValidation struct {
  Name                    string
  ContainerType           ContainerType
  Error                   error
  FileChecksumValidations ContainerChecksumValidation
  Subcontainers           []*ContainerValidation
}

func Validate(ctx context.Context, sessionRecordingId string, f storage.FS, keyUnwrapFn kms.KeyUnwrapCallbackFunc) (*Validation, error) {
  // implementation of validate
}
```

Note that this eliminates the `FailedChecksums` map. I think this is not necessary for the bsr package to track directly. We could provide a method on `Validation` to extract just the failed checksums, but I think we can also leave that as something for the caller of the bsr package to do.